StylePropertyMap restructure

njlawton opened the pull request:
Restructuring the current StylePropertyMap classes to be consistent with the original spec

Commit:
Created class shell for the InlineStylePropertyMap class
Also created a shell for its test class

Commit:
Added InlineStyleProportyMap class and test class to the traget config file
Currently this causes the test file to output an error. Did this break because the class is too barebones to be considered a class or is it something to do with the syntax of the class. I am especially scepticle that I have made a mistake in lines 29 - 31

Commit:
Fixed test case running error
Problem was due to missing a comma between strings in the target-loader.js file

Commit:
Implemented constructor for the InlineStylePropertyMap class including tests
This Involved two steps firstly adding a method to the prototype of the Element class called styleMap witch returns the InlineStylePropertyMap object for that class and then secondly writing the actual constructor for the class

Commit:
Changed class structure for StyleProportyMap
Structure changed to reflect the original spec layed out for StyleProportyMap

Commit:
Removed obsolete files related to the change of structure in StyleProportyMap
Removed InlineStylePropertyMap class as well as adding some minor changes and updates to configuration files

Commit:
Fixed small formatting errors found using gjslint

wilddamon commented on src/style-property-map-readonly.js:
```diff
@@ -14,27 +14,34 @@
 
-  ComputedStylePropertyMap.prototype.append = function(property, value) {
-    throw new TypeError('ComputedStylePropertyMap is immutable');
+  StylePropertyMapReadOnly.prototype.append = function(property, value) {
```
The modification functions should just not exist on StylePropertyMapReadOnly - they can just be added to StylePropertyMap (and the error can be "Not implemented yet" now instead).

wilddamon commented on test/js/inline-style-property-map.js:
```diff
@@ -0,0 +1,16 @@
+suite('Inline StylePropertyMap', function() {
+  setup(function() {
+    this.element = document.createElement('div');
+    document.documentElement.appendChild(this.element);
```
It's more common to write

document.body.appendChild(element)

Otherwise the element ends up going after the body, which is a little bit strange.

wilddamon commented on test/js/inline-style-property-map.js:
```diff
@@ -0,0 +1,16 @@
+
+  test('Test that a new InlineStylePropertyMap is created when a HTML element referance calls the ' +
+	    'styleMap method', function() {
+	 var inlineStyleMap = this.element.styleMap();
```
Fix indenting, should only be 2 spaces in from test(.

wilddamon commented on src/style-property-map.js:
```diff
@@ -14,16 +14,18 @@
-    has: function(property) {},
-    set: function(property, value) {},
-    getProperties: function() {}
+  StylePropertyMap.prototype.prototype = StylePropertyMapReadOnly.prototype;
```
Right now you're adding stuff to the ReadOnly one's prototype too. I have some ideas about how to fix the prototype chain which I'll look into as well, but in the meantime it's best to make a copy.

rjwright replied:
This is what I told him to do. But it was actually wrong, it should have been 

```
StylePropertyMap.prototype.__proto__ = StylePropertyMapReadOnly.prototype;
```

Sorry Nick.

But using the inherit method is neater anyway.

Commit:
Made changes StylePropertyMap class to fix problems that arose in the pull request

Commit:
Fixed small issues in Inline StylePropertyMaptest script

wilddamon commented on src/style-property-map-readonly.js:
```diff
@@ -14,27 +14,26 @@
-  ComputedStylePropertyMap.prototype.delete = function(property) {
-    throw new TypeError('ComputedStylePropertyMap is immutable');
+  StylePropertyMapReadOnly.prototype = {
+    append: function(property, value) {},
```
This wasn't quite what I meant. The StylePropertyMapReadOnly should not have any of the mutating methods. 
append, delete and set should all be moved into StylePropertyMap, and this map can be removed.

wilddamon commented on test/js/computed-style-property-map.js:
```diff
@@ -8,16 +8,9 @@ suite('ComputedStylePropertyMap', function() {
+  test('Computed StylePropertyMap.get works for properties that can only be ' +
       'NumberValues', function() {
-    var computedStyleMap = new ComputedStylePropertyMap(this.element);
+    var computedStyleMap = new StylePropertyMapReadOnly(getComputedStyle(this.element));
```
Please use the new getComputedStyleMap function instead

wilddamon commented on test/js/inline-style-property-map.js:
```diff
@@ -0,0 +1,16 @@
+    document.body.removeChild(this.element);
+  });
+
+  test('Test that a new InlineStylePropertyMap is created when a HTML element referance calls the ' +
```
This test seems suspicious - there is no InlineStylePropertyMap anymore? Perhaps it should be "Test that element.styleMap returns an Inline StylePropertyMap"

Also the formatting in this function is still weird...

Commit:
Fixed issues that came up in the pull request

Commit:
Merge branch 'master' into Style-map-restructure

Commit:
Fixed errors in StylePropertyMap due to merging with master branch

Commit:
Small chenges to formatting in the Inline StylePropertyMap test class
Changes due to problems caught by the linter

njlawton commented:
Should be good now

wilddamon commented on src/style-property-map-readonly.js:
```diff
@@ -103,27 +92,26 @@
 
-  ComputedStylePropertyMap.prototype.set = function(property, value) {
-    throw new TypeError('ComputedStylePropertyMap is immutable');
+  StylePropertyMapReadOnly.prototype.set = function(property, value) {
```
You can delete set now too.

Commit:
Removed set method from the StylePropertyMapReadOnly class
After the current class restructure this method should not be presesnt in this class and has therefore been removed

rjwright commented:
So just to be clear, the purpose of this CL is to
- remove ComputedStylePropertyMap and move its logic into StylePropertyMapReadOnly
- make StylePropertyMap instantiable and use it as the return type for Element.styleMap (the inline style map)
- make StylePropertyMap extend StylePropertyMapReadOnly
- make the StylePropertyMap/StylePropertyMapReadOnly constructors take a style object as an argument

right?

rjwright commented on test/js/inline-style-property-map.js:
```diff
@@ -0,0 +1,15 @@
+    document.body.removeChild(this.element);
+  });
+
+  test('Test that a new InlineStylePropertyMap is created when a HTML element referance calls the styleMap method', function() {
```
s/InlineStylePropertyMap/StylePropertyMap

I would change this to "Element.styleMap returns a StylePropertyMap."

rjwright commented on test/js/inline-style-property-map.js:
```diff
@@ -0,0 +1,15 @@
+
+  test('Test that a new InlineStylePropertyMap is created when a HTML element referance calls the styleMap method', function() {
+    var inlineStyleMap = this.element.styleMap();
+    assert.instanceOf(inlineStyleMap, StylePropertyMap, 'The styleMap method should return an instance of InlineStylePropertyMap');
```
s/InlineStylePropertyMap/StylePropertyMap

rjwright commented on test/js/computed-style-property-map.js:
```diff
@@ -8,16 +8,9 @@ suite('ComputedStylePropertyMap', function() {
-  });
-
-  test('ComputedStylePropertyMap.get works for properties that can only be ' +
+  test('Computed StylePropertyMap.get works for properties that can only be ' +
```
Wording here is a little odd. Maybe
"Computed StylePropertyMapReadOnly.get works for properties that can only be NumberValues"
or
"Getting opacity from the computed style map works as expected"

rjwright commented:
LGTM with comments

rjwright commented on src/style-property-map-readonly.js:
```diff
@@ -103,27 +92,22 @@
-  function getComputedStyleMap(element) {
-    return new ComputedStylePropertyMap(element);
-  }
+  scope.getComputedStyleMap = function(element) {
```
Please make this 

```
getComputedStyleMap = function(element)
```

and put

```
scope.getComputedStyleMap = getComputedStyleMap
```

below (with the other scope member assignments).

rjwright commented on src/style-property-map-readonly.js:
```diff
@@ -103,27 +92,22 @@
+  internal.StylePropertyMapReadOnly = StylePropertyMapReadOnly;
   if (TYPED_OM_TESTING)
-    testing.ComputedStylePropertyMap = ComputedStylePropertyMap;
+    testing.StylePropertyMapReadOnly = StylePropertyMapReadOnly;
```
One line blocks should have braces.

rjwright commented on src/style-property-map.js:
```diff
@@ -14,16 +14,25 @@
+    throw new TypeError('Function not implemented yet');
+  };
+
+  scope.Element.prototype.styleMap = function() {
```
Can just be

```
Element.prototype.styleMap
```

Since we haven't explicitly put Element on scope.

Commit:
Changed wording in the Computed and Inline StylePropertyMap test classes

rjwright commented on test/js/computed-style-property-map.js:
```diff
@@ -8,25 +8,17 @@ suite('ComputedStylePropertyMap', function() {
   });
 
-  test('window.getComputedStyleMap works', function() {
+  test('window.getComputedStyleMap method functions correctly', function() {
```
This was worded better before.

Commit:
Adressed issues brought up in pull request

Commit:
Reversed changed in wording that was done to a test case in a previous commit

njlawton commented:
Yes what you stated above were the purposes of the CL

rjwright commented on test/js/inline-style-property-map.js:
```diff
@@ -0,0 +1,15 @@
+    document.body.removeChild(this.element);
+  });
+
+  test('Test that the Element.styleMap method returns a StylePropertyMap object', function() {
```
'Test that the' is redundant. Remember there are dozens of test cases.

rjwright commented on test/js/inline-style-property-map.js:
```diff
@@ -0,0 +1,15 @@
+
+  test('Test that the Element.styleMap method returns a StylePropertyMap object', function() {
+    var inlineStyleMap = this.element.styleMap();
+    assert.instanceOf(inlineStyleMap, StylePropertyMap, 'The styleMap method should return an instance of InlineStylePropertyMap');
```
s/InlineStylePropertyMap/StylePropertyMap

rjwright commented on test/js/computed-style-property-map.js:
```diff
@@ -8,16 +8,8 @@ suite('ComputedStylePropertyMap', function() {
-  test('ComputedStylePropertyMap.get works for properties that can only be ' +
-      'NumberValues', function() {
-    var computedStyleMap = new ComputedStylePropertyMap(this.element);
+  test('Computed StylePropertyMap.get works for properties that can be NumberValues', function() {
```
Wording here is a little odd. This is testing a method of StylePropertyMapReadOnly, not StylePropertyMap.

Maybe
"Computed StylePropertyMapReadOnly.get works for properties that can only be NumberValues"
or
"Getting opacity from the computed style map works as expected"

njlawton commented:
should "Computed StylePropertyMapReadOnly.get works for properties that can only be NumberValues"
just be "Computed StylePropertyMapReadOnly.get works for properties that can be NumberValues"

this was no my test but I think what it is trying to say is that currently NumberValues works and other values will work later

rjwright commented:
> should "Computed StylePropertyMapReadOnly.get works for properties that can only be NumberValues"
> just be "Computed StylePropertyMapReadOnly.get works for properties that can be NumberValues"

Yep :)

Commit:
Fixed issues brought up in pull request

rjwright commented on test/js/keyword-value.js:
```diff
@@ -33,8 +33,7 @@ suite('KeywordValue', function() {
 
-  test('KeywordValue.cssString and KeywordValue.keywordValue are correctly ' +
-      'initialized', function() {
+  test('KeywordValue.cssString and KeywordValue.keywordValue are correctly initialized', function() {
```
Is this  meant to be in this PR?
